### ISSUE TITLE
Have a more extensive clippy help message.

### ISSUE THREAD

**Issue by dogunbound** [rust-lang/rust-clippy#10436]
### Description

Currently, clippy only shows "Common options". There should be a way to show ALL possible arguments that can be passed into clippy, even if that means linking it to a website.

Maybe I don't know the command, or where to find all of the clippy arguments. My issue was I knew that there was a way to run clippy through examples, but didn't know the argument. Luckily, I guess `--examples`, but it would have been much easier if there was a decent way to figure this stuff out. 

I genuinely think this help message is useless:
```
[dogunbound@archlinux ~]$ cargo clippy --help
Checks a package to catch common mistakes and improve your Rust code.

Usage:
    cargo clippy [options] [--] [<opts>...]

Common options:
    --no-deps                Run Clippy only on the given crate, without linting the dependencies
    --fix                    Automatically apply lint suggestions. This flag implies `--no-deps`
    -h, --help               Print this message
    -V, --version            Print version info and exit
    --explain LINT           Print the documentation for a given lint

Other options are the same as `cargo check`.

To allow or deny a lint from the command line you can use `cargo clippy --`
with:

    -W --warn OPT       Set lint warnings
    -A --allow OPT      Set lint allowed
    -D --deny OPT       Set lint denied
    -F --forbid OPT     Set lint forbidden

You can use tool lints to allow or deny lints from your code, eg.:

    #[allow(clippy::needless_lifetimes)]
```

### Version

```text
[dogunbound@archlinux ~]$ cargo clippy --version
clippy 0.1.67 (d5a82bb 2023-02-07)
```

**Comment by Alexendoo** [comment]
We could be more explicit in saying run `cargo check --help` to see the rest of the args rather than just referring to it

**Comment by dogunbound** [comment]
> We could be more explicit in saying run `cargo check --help` to see the rest of the args rather than just referring to it

If you mean, `cargo clippy -h` gives the short help message, and `cargo clippy --help` gives the longer help message, that would be great. Some other applications (I don't remember off the top of my head) already do that. As long as there is some way to reference all of the possible arguments.

**Comment by dogunbound** [comment]
> I meant do something like
> 
> ```diff
> -Other options are the same as `cargo check`.
> +For the other options see `cargo check --help`.
> ```
> 
> Listing the options from `cargo check` in the clippy help output would be even better, but that's tricky because we don't actually parse those flags we just pass them on to `cargo check`

Oh yeah. that's way clearer.

**Comment by dogunbound** [comment]
> 

I know EXACTLY what to do:

```
Common options:
    --no-deps                Run Clippy only on the given crate, without linting the dependencies
    --fix                    Automatically apply lint suggestions. This flag implies `--no-deps`
    -h, --help               Print this message
    -V, --version            Print version info and exit
    --explain LINT           Print the documentation for a given lint

    cargo check --help   Print all of the available arguments clippy can accept.
```
